chia/server/class_for_type.py

```diff
@@ -0,0 +1,38 @@
+from __future__ import annotations
+
+from typing import Any
+
+from chia.server.outbound_message import NodeType
+from chia.util.class_registry import ClassRegistry
+
+
+GLOBAL_REGISTRY = ClassRegistry()
+
+
+def register_class(
+    node_type: Any,
+    cls: Any,
+) -> None:
+    GLOBAL_REGISTRY.register_class(node_type, cls)
+
+
+def class_for_type(
+    node_type: Any,
+) -> Any:
+    return GLOBAL_REGISTRY.class_for_type(node_type)
+
+
+def register_all():
```

Comment:
This moves the problem to this function, which can be isolated to something at a higher level than `server`, since ideally `server` shouldn't have to know about particular types of servers.

Reply:
So in other words, this PR as it is doesn't quite solve the cycle problem. But it changes it to "where early on in execution does it make sense to call `register_all`?"